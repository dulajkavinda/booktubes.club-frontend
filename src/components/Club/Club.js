import * as React from "react";
import { styled } from "@mui/material/styles";
import Card from "@mui/material/Card";
import CardHeader from "@mui/material/CardHeader";
import CardContent from "@mui/material/CardContent";
import CardActions from "@mui/material/CardActions";
import Avatar from "@mui/material/Avatar";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";
import { red } from "@mui/material/colors";
import ShareIcon from "@mui/icons-material/Share";
import CardMedia from "@mui/material/CardMedia";
import Button from "@mui/material/Button";

import "./Club.scss";

export default function Club({ club, type }) {
  const [open, setOpen] = React.useState(false);
  const handleOpen = () => setOpen(true);

  return (
    <Card className="club_main" sx={{ width: 345 }}>
      <CardHeader
        avatar={
          <Avatar sx={{ bgcolor: red[500] }} aria-label="recipe">
            R
          </Avatar>
        }
        title={club.club_name}
        subheader={club.category[0]}
      />

      <CardContent>
        <Typography variant="body2" color="text.secondary">
          {type === "joined" ? (
            <div className="card_media_main">
              <CardMedia
                component="img"
                height="220px"
                image="https://bookaudio.online/uploads/images/media/125/usplBPlBuHH5yS9m-charles-dickens-oliver-twist.jpg"
                alt="Oliver Twist"
              />
              <div className="book_summary">
                <div>
                  The novel follows the journey of the titular character, Oliver
                  Twist. Oliver, an orphan since birth, spends much of his
                  childhood at a “child farm” (orphanage) with too many children
                  and too little food. The farm is located roughly 70 miles
                  outside London.
                </div>
                <div className="progress">
                  <span>Progress : 90%</span>
                </div>
              </div>
            </div>
          ) : (
            club.description
          )}
        </Typography>
      </CardContent>
      <CardActions>
        {type === "joined" ? (
          <Button variant="contained">View Club</Button>
        ) : (
          <Button color="success" variant="contained">
            Join Club
          </Button>
        )}

        <IconButton aria-label="share">
          <ShareIcon />
        </IconButton>
      </CardActions>
    </Card>
  );
}
