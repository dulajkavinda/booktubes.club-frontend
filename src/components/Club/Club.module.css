.news {
  width: 400px;
  position: relative;
  display: flex;
  height: 300px;
  border-radius: 20px;
  margin: 20px;
  box-shadow: rgba(0, 0, 0, 0.16) 0px 3px 6px, rgba(0, 0, 0, 0.15) 0px 3px 6px;
  justify-content: space-between;
  cursor: pointer;
  overflow: hidden;
}
.news_corner {
  content: "";
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 40px 40px 0 0;
  opacity: 0.4;
  right: 0;
  bottom: 0;
  position: absolute;
}

.left_col {
  width: 40%;
  height: 300px;
}
.right_col {
  width: 60%;
  height: 300px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.news_title {
  color: #1a1c26;
  opacity: 0.6;
  font-size: 1.125rem;
  font-weight: 600;
  letter-spacing: -0.5px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 4; /* number of lines to show */
  -webkit-box-orient: vertical;
}

.date {
  color: #1a1c26;
  opacity: 0.6;
  font-size: 0.7rem;
  font-weight: 600;
  letter-spacing: -0.5px;
  margin-top: 10px;
}
.buttons {
  display: flex;
  flex-direction: row;
}
.buttons > div {
  margin-right: 10px;
}

.left_col > img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-bottom-left-radius: 20px;
  border-top-left-radius: 20px;
  opacity: 0.8;
}

.left_col > img:before {
  background: #f1f1f1;
  content: "\1F517"" broken image :(" attr(alt);
  left: 0;
  height: 100%;
  width: 40%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  position: absolute;
}

.modal_window {
  width: 400px;
  height: 300px;
  position: absolute;

  transform: translate(-50%, -50%);
  padding: 2em;
  background: black;
  background-color: white;
}

.news_container {
  margin-top: 5vh;
  margin-bottom: 5vh;
  display: flex;
  width: 100%;

  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
}

.heading {
  color: #252424;
  opacity: 0.75;
  font-size: 1.125rem;
  font-weight: 600;
  letter-spacing: -0.5px;
  line-height: 20px;
  display: flex;
  justify-content: center;
}

.modal_content {
  display: flex;
  justify-content: center;
  flex-direction: column;
}

.modal_image {
  border-radius: 20px;
  margin-top: 10px;
  opacity: 0.8;
  width: 100%;
  max-height: 28vh;
  position: relative;
}

.modal_content > div > img:before {
  background: #f1f1f1;
  content: "\1F517"" broken image :( just like their promises" attr(alt);
  left: 0;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 16px;
  position: absolute;
}

.news_body {
  color: black;
  opacity: 0.6;
  font-size: 1rem;
  font-weight: 600;
  letter-spacing: -0.5px;
  margin-top: 20px;
  margin-bottom: 10px;
  max-height: 30vh;
  overflow: scroll;
}
.news_body_loading {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 20vh;
}
