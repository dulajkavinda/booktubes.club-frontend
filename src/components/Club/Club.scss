@import "../../styles/config";

.club_main {
  margin-right: 10px;
}

.card_media_main {
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  .progress {
    font-family: $primary-text-style;
    font-weight: 600;
  }

  .book_summary {
    margin-left: 10px;
  }
}
