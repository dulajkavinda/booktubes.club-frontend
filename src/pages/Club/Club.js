import React, { useState } from "react";
import { useLocation } from "react-router-dom";

export default function Club() {
  let location = useLocation();
  const [data, setData] = useState(location.state.club);

  return (
    <div style={{ display: "flex", justifyContent: "center" }}>
      <div
        style={{
          display: "flex",
          justifyContent: "center",

          height: "200px",
          width: "70%",
          flexDirection: "column",
          backgroundColor: "white",
        }}
      >
        <span style={{ fontSize: "20px" }}>Club Name: {data.club_name}</span>
        <span style={{ fontSize: "16px", marginBottom: "10px" }}>
          {data.description}
        </span>
        <span style={{ fontSize: "16px" }}>Category: {data.category[0]}</span>

        <span style={{ fontSize: "16px" }}>
          Total members :{data.members.length}
        </span>
        <span style={{ fontSize: "16px" }}>
          Total Posts: {data.posts.length}
        </span>
      </div>
      <div
        style={{
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          height: "100px",
          width: "30%",
          backgroundColor: "black",
        }}
      >
        {data.club_name}
      </div>
    </div>
  );
}
