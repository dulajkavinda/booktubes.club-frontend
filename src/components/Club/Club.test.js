import React from "react";
import { shallow, mount } from "enzyme";
import Club from "./Club";

import club_data from "../../data/club.json";

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Club club={club_data.clubs[0]} />);
});

describe("dashboard basic tests", () => {
  it("is snapshot changed", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
