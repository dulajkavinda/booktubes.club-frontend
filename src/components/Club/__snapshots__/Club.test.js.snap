// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard basic tests is snapshot changed 1`] = `
<ForwardRef(Card)
  className="club_main"
  sx={
    Object {
      "width": 345,
    }
  }
>
  <ForwardRef(CardHeader)
    avatar={
      <ForwardRef(Avatar)
        aria-label="recipe"
        sx={
          Object {
            "bgcolor": "#f44336",
          }
        }
      >
        R
      </ForwardRef(Avatar)>
    }
    subheader="Horror"
    title="Millers Club"
  />
  <ForwardRef(CardContent)>
    <ForwardRef(Typography)
      color="text.secondary"
      variant="body2"
    >
      This impressive paella is a perfect party dish and a fun meal to cook together with your guests. Add 1 cup of frozen peas along with the mussels, if you like.
    </ForwardRef(Typography)>
  </ForwardRef(CardContent)>
  <ForwardRef(CardActions)>
    <ForwardRef(Button)
      color="success"
      variant="contained"
    >
      Join Club
    </ForwardRef(Button)>
    <ForwardRef(IconButton)
      aria-label="share"
    >
      <Memo(ForwardRef(ShareIcon)) />
    </ForwardRef(IconButton)>
  </ForwardRef(CardActions)>
</ForwardRef(Card)>
`;
